src/hotkey/key_window.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <stdint.h>
+
+/*
+ * The key_window module provides a FIFO key_window interface to help with bookkeeping the
```

Comment:
I think the benefits of using ring_array do not significantly outweigh the drawbacks. Although it would provide a clean push/pop, using STAILQ is also a one liner in this aspect and is not significantly harder to understand. ring_array also lacks a foreach equivalent, which means we would need to write a messier loop for freeing the contents. Additionally, using ring_array still does not remove the need to check whether the the window is full. Furthermore, if we are to port this hotkey change to twemcache, twemcache already has mc_queue so it would be mostly read to go; however, were we to switch to ring_array we would need to port ring_array over to twemcache as well. For these reasons, I think we should keep it as it is for now.